lib/src/settings.rs

```diff
@@ -179,6 +179,19 @@ impl UserSettings {
             .get_string("ui.graph.style")
             .unwrap_or_else(|_| "curved".to_string())
     }
+
+    pub fn max_new_file_size(&self) -> Option<u64> {
+        let raw = self
+            .config
+            .get_int("snapshot.max-new-file-size")
+            .map(|x| x.max(0) as u64)
+            .unwrap_or(1024 * 1024);
```

Comment:
Out of scope: It's unfortunate that we specify a default both here and in `misc.toml`. Maybe once we have defined a typed config schema, we can read the value here and have the config schema know to give us the default if the user has set it to a non-integer value.

(To be clear, I think it's good to have it in both places for now. I like that `jj config list --include-defaults` lets you see most defaults, and I think it's good to use the default value here if the user had set e.g. `snapshot.max-new-file-size = "100kB"`.)

Reply:
This information is also duplicated in config-schema.json; I guess in an ideal world we generate that, or generate from that.

I'm not very happy about ill-typed config parameters getting silently defaulted, now that you mention it, but yeah that's a broader issue for future work.